apps/portals/e2e/explore.spec.ts

```diff
@@ -109,6 +114,109 @@ const getAvailableFacetsDiv = (page: Page) => {
     .filter({ hasText: 'Available Facets' })
 }
 
+const countVisibleElements = async (allCells: Locator, count: number) => {
+  let visibleCount = 0
+  for (let i = 0; i < count; i++) {
+    const cell = allCells.nth(i)
+    if (await cell.isVisible()) {
+      visibleCount++
+    }
+  }
+  return visibleCount
+}
+
+const toggleChartVisibilityAndVerifyCount = async (page: Page) => {
+  const chartsPanel = page.locator('.FacetNav')
+
+  const allChartCells = chartsPanel.getByRole('figure')
+  const allChartsCount = await allChartCells.count()
+
+  let visibleChartCount = 0
+  let firstVisibleindex = -1
+  for (let i = 0; i < allChartsCount; i++) {
+    const chart = allChartCells.nth(i)
+    if (await chart.isVisible()) {
+      visibleChartCount++
+    }
+    if ((await chart.isVisible()) && firstVisibleindex == -1) {
+      firstVisibleindex = i
+    }
+  }
+  const chartToTestIndex = firstVisibleindex
+  const initialCount = visibleChartCount
+
+  await test.step('confirm chart visibility and count', async () => {
+    const chartCell = allChartCells.nth(chartToTestIndex)
+    const chartToolbar = await chartCell.getByRole('toolbar')
+    await chartToolbar
+      .getByRole('button', { name: 'Hide graph under Show More' })
+      .click()
+
+    const newCount = await countVisibleElements(allChartCells, allChartsCount)
```

Comment:
Does this work? If so, maybe get rid of this function all together and just do this
```suggestion
    const newCount = await allChartCells.filter({ has: page.locator('visible') }).count()
```